File: l-2-4/ts/Expression.ts
import { ExpressionError } from "./ExpressionError";

export class Expression {
    private a: number;
    private b: number;

    constructor(a: number, b: number) {
        this.a = a;
        this.b = b;
    }

    public getValues(): [number, number] {
        return [this.a, this.b];
    }

    public calculate(): number {
        if (this.b === 0) throw new ExpressionError("Division by zero is not allowed");
        return this.a / this.b;
    }

    public getSqrtOfA(): number {
        if (this.a < 0) throw new ExpressionError("Square root of negative number is not allowed");
        return Math.sqrt(this.a);
    }
}

File: l-2-4/cs/Expression.cs
using System;

public class Expression
{
    private double a;
    private double b;

    public Expression(double a, double b)
    {
        this.a = a;
        this.b = b;
    }

    public (double, double) GetValues() => (a, b);

    public double Calculate()
    {
        if (b == 0)
            throw new ExpressionException("Ділення на нуль заборонено.");
        return a / b;
    }

    public double GetSqrtOfA()
    {
        if (a < 0)
            throw new ExpressionException("Корінь з від’ємного числа неможливий.");
        return Math.Sqrt(a);
    }
}
